packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -0,0 +1,916 @@
+})<{
+  disabled: boolean
+  selected: boolean
+}>`
+  text-align:left;
+  border: none;
+  background-color: ${({ theme, selected, disabled }) => {
+    if (selected) {
+      return `${theme.colors.primary.background};`
+    }
+    if (disabled) {
+      return `${theme.colors.neutral.backgroundDisabled};`
+    }
+
+    return `${theme.colors.neutral.background};`
+  }}
+  padding: ${({ theme }) => theme.space['1.5']}
+    ${({ theme }) => theme.space['2']} ${({ theme }) => theme.space['1.5']}
+    ${({ theme }) => theme.space['2']};
+  margin-left: ${({ theme }) => theme.space['0.5']};
+  margin-right: ${({ theme }) => theme.space['0.5']};
+  color:  ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.textDisabled}` : `${theme.colors.neutral.text}`)};
+
+  &:hover, :focus {
+    background-color: ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.backgroundStrongDisabled};` : `${theme.colors.primary.background};`)}
+    color: ${({ theme, disabled }) => (disabled ? `${theme.colors.neutral.textStrongDisabled}` : `${theme.colors.primary.text}`)};
+    cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+    outline: none;
+  }
+`
+const PopupFooter = styled.div`
+  width: 100%;
+  padding: ${({ theme }) => theme.space[1.5]} ${({ theme }) => theme.space[2]}
+    ${({ theme }) => theme.space[1.5]} ${({ theme }) => theme.space[2]};
+  box-shadow: ${({ theme }) => theme.shadows.dropdown};
+`
+const StyledCheckbox = styled(Checkbox)`
+  width: 100%;
+  position: static;
+  text-align: left;
+`
+const EmptyState = styled(Stack)`
+  padding: ${({ theme }) => theme.space[2]};
+`
+
+const moveFocusDown = () => {
+  const listOptions: ChildNode[] = []
+  const options = document.querySelectorAll('#items')
+  const activeItem = document.activeElement
+  options.forEach(val => {
+    listOptions.push(...Array.from(val.childNodes))
+
+    return null
+  })
+
+  if (listOptions) {
+    for (let i = 0; i < listOptions?.length; i += 1) {
+      const listLength = listOptions.length
+      if (
+        activeItem === listOptions[i] &&
+        activeItem !== listOptions[listLength - 1]
+      ) {
+        ;(listOptions[i + 1] as HTMLElement).focus()
+      }
+    }
+  }
+}
+const moveFocusUp = () => {
+  let listOptions: ChildNode[] = []
+  const options = document.querySelectorAll('#items')
+  const activeItem = document.activeElement
+  options.forEach(val => {
+    listOptions = listOptions.concat(Array.from(val.childNodes))
+
+    return null
+  })
+  if (listOptions) {
+    for (let i = 0; i < listOptions.length; i += 1) {
+      if (activeItem === listOptions[i] && activeItem !== listOptions[0]) {
+        ;(listOptions[i - 1] as HTMLElement).focus()
+      }
+    }
+  }
+}
+const handleKeyDownSelect = (key: string) => {
+  if (key === 'ArrowDown') {
+    moveFocusDown()
+  }
+  if (key === 'ArrowUp') {
+    moveFocusUp()
+  }
+}
+const HandleDropdown = (
```

Comment:
Its not a component better to call this handleDropdownRef or something lowerCamelCase